packages/expo-notifications/android/src/main/java/expo/modules/notifications/permissions/NotificationPermissionsModule.java

```diff
@@ -0,0 +1,75 @@
+package expo.modules.notifications.permissions;
+
+import android.app.NotificationManager;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.Promise;
+import org.unimodules.core.arguments.ReadableArguments;
+import org.unimodules.core.interfaces.ExpoMethod;
+import org.unimodules.interfaces.permissions.PermissionsStatus;
+
+import androidx.core.app.NotificationManagerCompat;
+
+import static org.unimodules.interfaces.permissions.PermissionsResponse.CAN_ASK_AGAIN_KEY;
+import static org.unimodules.interfaces.permissions.PermissionsResponse.EXPIRES_KEY;
+import static org.unimodules.interfaces.permissions.PermissionsResponse.GRANTED_KEY;
+import static org.unimodules.interfaces.permissions.PermissionsResponse.PERMISSION_EXPIRES_NEVER;
+import static org.unimodules.interfaces.permissions.PermissionsResponse.STATUS_KEY;
+
+public class NotificationPermissionsModule extends ExportedModule {
+  private static final String EXPORTED_NAME = "ExpoNotificationPermissionsModule";
+
+  private static final String ANDROID_RESPONSE_KEY = "android";
+  private static final String IMPORTANCE_KEY = "importance";
+  private static final String INTERRUPTION_FILTER_KEY = "interruptionFilter";
+
+  public NotificationPermissionsModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return EXPORTED_NAME;
+  }
+
+  @ExpoMethod
+  public void getPermissionsAsync(final Promise promise) {
+    promise.resolve(getPermissionsBundle());
+  }
+
+  @ExpoMethod
+  public void requestPermissionsAsync(@SuppressWarnings("unused") final ReadableArguments permissionsTypes, final Promise promise) {
```

Comment:
I was trying to specify a clear native module interface 
```ts
export interface NotificationPermissionsModule extends ProxyNativeModule {
  getPermissionsAsync: () => Promise<NotificationPermissionsStatus>;
  requestPermissionsAsync: (
    request: NotificationPermissionsRequest
  ) => Promise<NotificationPermissionsStatus>;
}
```
and make all the native modules conform to it.

This way the TS code is simpler imho, otherwise we would have to write something like
```ts
export async function requestPermissionsAsync(permissions?: NotificationPermissionsRequest) {
  if (Platform.OS === "android") {
    // Android doesn't support requesting permissions
    return await getPermissionsAsync();
  }

  return await NotificationPermissionsModule.requestPermissionsAsync(
    permissions ?? {
      allowAlert: true,
      allowBadge: true,
      allowSound: true,
    }
  );
}
```

Maybe this would be clear too?